homeassistant/components/whirlpool/sensor.py

```diff
@@ -89,6 +113,23 @@ def washer_dryer_state(washer_dryer: WasherDryer) -> str | None:
     return WASHER_DRYER_MACHINE_STATE.get(machine_state)
 
 
+def oven_state(machine: Oven) -> str | None:
+    """Determine correct states for an oven."""
+
+    if machine.get_door_opened():
```

Comment:
I am considering removing the door open state from the washer/dryer states, into its own binary_sensor (but I'm still trying to get feedback from someone who has a washer).
For oven, I also suggest to remove this and put it in a binary sensor (in a future PR). I will add the binary_sensor platform soon, so it would be easier to add a new one after.